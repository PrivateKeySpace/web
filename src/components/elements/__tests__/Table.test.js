import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { Table } from '..'

const testColumns = [
  {name: 'Date', key: 'date'},
  {name: 'Status', key: 'status'},
  {name: 'Created By', key: 'createdBy'},
  {name: 'To', key: 'to'},
  {name: 'Amount', key: 'amount'},
  {name: 'Reason', key: 'reason'}
]

const testData = [
  {
    date: '23.05.2012',
    status: 'Request 2/3',
    createdBy: 'Mark Spenser',
    to: 'Bitcoin Address',
    amount: 100,
    reason: 'Test'
  },
  {
    date: <div>'23.05.2012'</div>,
    status: <div>'Request 2/3'</div>,
    createdBy: <i>'Mark Spenser'</i>,
    to: <i>'Bitcoin Address'</i>,
    amount: <b>100</b>,
    reason: <b>'Test'</b>
  },
  {
    date: '23.05.2012',
    createdBy: 'Mark Spenser',
    amount: 100,
    reason: 'Test'
  }
]

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders empty table without crashing and matches snapshot', () => {
  const table = render(
    <Table />
  )
  expect(table).toMatchSnapshot()
})

it('renders empty table with columns and default empty message, does not crash and matches snapshot', () => {
  const table = render(
    <Table columns={testColumns} />
  )
  expect(table).toMatchSnapshot()
})

it('renders empty table with columns and empty message, does not crash and matches snapshot', () => {
  const table = render(
    <Table columns={testColumns} emptyMessage='Test data is empty' />
  )
  expect(table).toMatchSnapshot()
})

it('renders table without columns, with data and empty message, does not crash and matches snapshot', () => {
  const table = render(
    <Table data={testData} emptyMessage='Test data is empty' />
  )
  expect(table).toMatchSnapshot()
})

it('renders table with columns, data and empty message, does not crash and matches snapshot', () => {
  const table = render(
    <Table columns={testColumns} data={testData} emptyMessage='Test data is empty' />
  )
  expect(table).toMatchSnapshot()
})
