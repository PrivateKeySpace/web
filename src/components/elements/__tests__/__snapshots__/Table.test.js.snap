// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty table with columns and default empty message, does not crash and matches snapshot 1`] = `
<table
  class="width100_zdxht7-o_O-table_zuaglh"
>
  <thead>
    <tr>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Date
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Status
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Created By
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        To
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Amount
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Reason
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        colspan="6"
      >
        <div
          class="width100_zdxht7-o_O-flex_jro6t0-o_O-justifyContentCenter_agrn11-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-noRows_a5gpv3"
        >
          No data
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders empty table with columns and empty message, does not crash and matches snapshot 1`] = `
<table
  class="width100_zdxht7-o_O-table_zuaglh"
>
  <thead>
    <tr>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Date
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Status
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Created By
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        To
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Amount
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Reason
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        colspan="6"
      >
        <div
          class="width100_zdxht7-o_O-flex_jro6t0-o_O-justifyContentCenter_agrn11-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-noRows_a5gpv3"
        >
          Test data is empty
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders empty table without crashing and matches snapshot 1`] = `
<table
  class="width100_zdxht7-o_O-table_zuaglh"
>
  <thead>
    <tr>
      
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        colspan="0"
      >
        <div
          class="width100_zdxht7-o_O-flex_jro6t0-o_O-justifyContentCenter_agrn11-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-noRows_a5gpv3"
        >
          No data
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders table with columns, data and empty message, does not crash and matches snapshot 1`] = `
<table
  class="width100_zdxht7-o_O-table_zuaglh"
>
  <thead>
    <tr>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Date
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Status
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Created By
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        To
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Amount
      </th>
      <th
        class="fontWeight400_tt122m-o_O-textAlignLeft_e4x16a-o_O-textTransformUppercase_4efw5a-o_O-px2_6ebohg-o_O-th_4acr85"
      >
        Reason
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        23.05.2012
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Request 2/3
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Mark Spenser
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Bitcoin Address
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        100
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Test
      </td>
    </tr>
    <tr>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <div>
          '23.05.2012'
        </div>
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <div>
          'Request 2/3'
        </div>
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <i>
          'Mark Spenser'
        </i>
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <i>
          'Bitcoin Address'
        </i>
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <b>
          100
        </b>
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        <b>
          'Test'
        </b>
      </td>
    </tr>
    <tr>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        23.05.2012
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Mark Spenser
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        100
      </td>
      <td
        class="px2_6ebohg-o_O-py4_r2hutn-o_O-td_1f1ffwf"
      >
        Test
      </td>
    </tr>
  </tbody>
</table>
`;
